src/modules/billing/service/charge-version-year.js

```diff
@@ -0,0 +1,64 @@
+const chargeProcessor = require('./charge-processor');
+const { omit } = require('lodash');
+const { logger } = require('../../../logger');
+const repository = require('../../../lib/connectors/repository');
+const { Batch } = require('../../../lib/models');
+const { assert } = require('@hapi/hoek');
+
+/**
+ * Given a Batch instance, writes all the invoices within the batch
+ * to the water.billing_invoices table
+ * @param {Batch} batch
+ * @return {Promise} resolves when all records written
+ */
+const createBatchInvoices = batch => {
+  assert(batch instanceof Batch, 'Batch expected');
+  const data = batch.invoices.map(invoice => ({
+    invoice_account_id: invoice.invoiceAccount.id,
+    invoice_account_number: invoice.invoiceAccount.accountNumber,
+    address: omit(invoice.address.toObject(), 'id'),
+    billing_batch_id: batch.id
+  }));
+  const tasks = data.map(row => repository.billingInvoices.create(row));
+  return Promise.all(tasks);
+};
+
+/**
+ * Given a charge version year record from the water.billing_batch_charge_version_years,
+ * processes the charge version for that year, and then populates a Batch model with
+ * related entities ready for persistence to DB
+ * @param {Object} chargeVersionYear
+ * @return {Batch}
+ */
+const createBatchFromChargeVersionYear = async chargeVersionYear => {
+  const {
+    charge_version_id: chargeVersionId,
+    financial_year_ending: financialYearEnding,
+    billing_batch_id: batchId
+  } = chargeVersionYear;
+
+  // Process charge data
+  const { error, data } = await chargeProcessor.processCharges(financialYearEnding, chargeVersionId);
+
+  if (error) {
+    const msg = 'Error processing charge version year';
+    logger.error(msg, chargeVersionYear);
```

Comment:
`logger.error` would take the error as the second arg. So this would be:

`logger.error(msg, error, { chargeVersionYear });`

Reply:
`error` here is a string so I re-jigged this to create an error object and send this to logger.error.
Should be fixed